src/query/shorthand.ts

```diff
@@ -141,7 +134,12 @@ export function fieldDef(encQ: EncodingQuery,
   // TODO: axis, legend
   for (const nestedPropParent of [Property.SCALE, Property.SORT]) {
     if (include[nestedPropParent]) {
-      if (encQ[nestedPropParent] && !isEnumSpec(encQ[nestedPropParent])) {
+      if (typeof encQ[nestedPropParent] === 'string') {
```

Comment:
1) Using isString from datalib for consistency
2) Put it below this clause.  You wanna split between when these properties really have nested values and when it's not rather than having the nested case in between.